### DISCUSSION TITLE
Is `-X build` supposed to re-run every time?

### DISCUSSION THREAD

**Discussion by harrymander** [tectonic-typesetting/tectonic#1015] [Q&A]
Not sure if this is a bug or expected, but `tectonic -X build` re-runs TeX every time, regardless of whether any files have changed.

E.g. using a project created with `tectonic -X new` and running `tectonic -X build` gives the same output everytime, even if a file is not changed:

```
note: "version 2" Tectonic command-line interface activated
Running TeX ...
Rerunning TeX because "default.aux" changed ...
Running xdvipdfmx ...
Writing `/home/harry/scratch/tectonic-test/build/default/default.pdf` (2.96 KiB)
Skipped writing 1 intermediate files (use --keep-intermediates to keep them)
```

Using `--keep-intermediates` gives the same result (but with the added 'writing default.aux' line).

I would think it should detect if any of the input files have changed and only build if they have?

**Comment by zzamboni** [comment]
I had the same idea, that `build` would only rerun things as needed, but it seems to always rebuild everything, even if no files have changed.

This also makes `watch` less useful, since it triggers a rebuild on any changes in the directory, even if no changes to the documents themselves have happened.
